File: algorithms/search/linear-search/linear-search.go
package main

import "fmt"

func linearSearch(list []int, target int) int {
	for i, value := range list {
		if value == target {
			return i
		}
	}
	return -1
}

func main() {
	list := []int{10, 20, 30, 40, 50}

	fmt.Println(linearSearch(list, 0))
	fmt.Println(linearSearch(list, 10))
	fmt.Println(linearSearch(list, 50))
}

File: algorithms/search/linear-search/linear-search.ts
function linearSearch<T>(list: T[], target: T): number {
  for (let i = 0; i < list.length; i++) {
    if (list[i] === target) {
      return i 
    }
  }
  return -1
}

const linearSearchList = [10, 20, 30, 40, 50]

console.log(linearSearch(linearSearchList, 0))
console.log(linearSearch(linearSearchList, 10))
console.log(linearSearch(linearSearchList, 50))